### PR TITLE
sundials: add v7.1.1

### PR DESCRIPTION
**Pull request by balos1**

### CODE REVIEW

File: var/spack/repos/builtin/packages/sundials/package.py
Comment (gardner48) [inline var/spack/repos/builtin/packages/sundials/package.py]: Any particular reason for not using the `sha256` like prior versions?

Comment (balos1) [reply var/spack/repos/builtin/packages/sundials/package.py]: 1) `spack checksum` cannot seem to find v7.1.0. 2) This will ensure any pulls via Spack count as a 'clone' in our stats instead of a download.

Comment (balos1) [reply var/spack/repos/builtin/packages/sundials/package.py]: I would like to go update all the older versions to use this approach too, but I will in a different PR.

**Review by gardner48 — DISMISSED** [review]

**Review by eugeneswalker — DISMISSED** [review]

**Comment by eugeneswalker** [comment]
The only failing CI build is `sundials %oneapi@2024.0.0`:
* [ sundials@7.1.0 /usmkaoz %oneapi@2024.0.0 arch=linux-ubuntu22.04-x86_64_v3 E4S OneAPI ](https://gitlab.spack.io/spack/spack/-/jobs/11656579)
```
...
 >> 1992    /tmp/root/spack-stage/spack-stage-sundials-7.1.0-usmkaozpsqwln5nce
             s5nsfch7vwcmeue/spack-src/src/nvector/sycl/nvector_sycl.cpp:873:6:
              error: conflicting types for 'N_VSpace_Sycl'
     1993      873 | void N_VSpace_Sycl(N_Vector X, long int* lrw, long int* li
             w)
     1994          |      ^
     1995    /tmp/root/spack-stage/spack-stage-sundials-7.1.0-usmkaozpsqwln5nce
             s5nsfch7vwcmeue/spack-src/include/nvector/nvector_sycl.h:116:22: n
             ote: previous declaration is here
     1996      116 | SUNDIALS_EXPORT void N_VSpace_Sycl(N_Vector v, sunindextyp
             e* lrw,
     1997          |                      ^
     1998    1 error generated.
```

@rscohn2

**Comment by balos1** [comment]
@eugeneswalker I added a patch that should fix the issue. However, I would like to release a 7.1.1 in place of 7.1.0 with patch, so dont merge this PR yet.  CC @gardner48.

**Review by eugeneswalker — APPROVED** [review]
Thanks!

**Comment by eugeneswalker** [comment]
Should we wait for further comment before merging?

**Comment by balos1** [comment]
Its good to merge now

**Pull request merged**
